### ISSUE TITLE
Data builders: generate builders for unknown types

### ISSUE THREAD

**Issue by BoD** [apollographql/apollo-kotlin#4603]
### Use case

Currently the data builders are only generated for all types known in the schema at build time.

When testing the case of receiving a new type (server's schema is more recent than the schema used at build time), it can be handy to also have the ability to build unknown types. With the responseBased codegen that would be the `Other` types of the generated sealed interfaces.

### Describe the solution you'd like

Have the codegen generate builders for the interfaces, taking the `__typename` as a parameter.

**Comment by martinbonnin** [comment]
Possible implementation:

```kotlin
    val data = GetNodeQuery.Data {
      // use 'Other' in the name
      // the typename is mandatory because we can't get it from the schema
      node = buildOtherNode("SomeNewType") {
        id = "42"
      }
    }
```

**Comment by martinbonnin** [comment]
This is now available in [version 3.7.4](https://github.com/apollographql/apollo-kotlin/releases/tag/v3.7.4)

**Comment by leinardi** [comment]
Awesome, thanks! :+1:
